src/Illuminate/Http/Response.php

```diff
@@ -82,11 +82,14 @@ public function setContent(mixed $content): static
      */
     protected function shouldBeJson($content)
     {
-        return $content instanceof Arrayable ||
-               $content instanceof Jsonable ||
-               $content instanceof ArrayObject ||
-               $content instanceof JsonSerializable ||
-               is_array($content);
+        return match (true) {
```

Comment:
Not sure if this is a real improvement for the readability..

Reply:
The match expression provides a clearer delineation between the different conditions and their outcomes, which could make it easier for developers to understand the logic at a glance. The goal was to simplify and streamline the conditional checks for better maintenance and understanding in the long run.

Reply:
I would argue it harder to read now.